query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -154,33 +155,39 @@ private String convertFilterToString(
           builder.append(convertExpressionToString(rhs, paramsBuilder, executionContext));
           builder.append(")");
           break;
+        case NOT_CONTAINS_KEY:
+          List<LiteralConstant> kvp = convertExpressionToMapLiterals(filter.getRhs());
+          builder.append(convertExpressionToMapKeyColumn(filter.getLhs()));
+          builder.append(" != ");
+          builder.append(convertLiteralToString(kvp.get(MAP_KEY_INDEX), paramsBuilder));
```

Comment:
CONTAINS_KEY and NOT_CONTAINS_KEY can really just have the same logic - assuming we translate the operators in `convertOperatorToString`, the only difference between these and the default case is the conversion of the LHS (using `convertExpressionToMapKeyColumn` instead of `convertExpressionToString`)

Reply:
resolved